doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+SAI IPv6 Segment Routing Update
+
+## Overview ##
+
+Segment Routing with IPv6(SRv6) support was added in SAI 1.2 about 4 years ago. Since then SAI 
+pipeline model for tunnels and MPLS fowarding has evolved and there have been several 
+modifications to SRv6 RFCs. This proposal will bring SRv6 APIs and attributes in-line with the 
+latest RFCs and SAI MPLS/tunnel pipeline model. 
+
+Note that SRv6 features and use cases continue to evolve and this proposal is focused on functions 
+and behaviors described in the SRv6 Network Programming RFC [
+Segment Routing over IPv6 (SRv6) Network Programming]. We will submit additional 
+PRs in future to add support for other functions and use cases.
+
+Following list of changes are being proposed:
+
+## Behavioral Model Changes ##
+
+### Endpoint Behavior ###
+In the earlier proposal, In order to program the hardware for an Endpoint SID, first a route entry is created for the local SID and then a nexthop entry 
+is created to specify the endpoint behavior and parameters. This will require either the application or the operating system responsible for making the SAI 
+API calls to create two additional objects for every LocalSID entry - a route object and a nexthop object. This model is different from the programming model 
+used for IP tunnels and MPLS. For IP and MPLS tunnel terminations, a separate tunnel_term or Insegment object is created without a need to decompose the tunnel 
+termination entry into multiple SAI objects. 
+This proposal will follow the programming model of MPLS segment routing and IP tunnel termination by adding a LocalSID object for SRv6 similar to InSegment
+for MPLS.
+  
+![SRv6 Endpoint Behavioral Model](figures/SRv6_Endpoint_behavioral_model.png "Figure 1: Endpoint Behavior ")
+__Figure 1: Endpoint Behavior.__
+
+### Source/Headend Behavior ###
+- This proposal will remove "Endpoint" from the Nexthop table attributes and leave other attributes like SIDLIST in order to implement the Headend Behavior.
+
+![SRv6 Headend Behavioral Model](figures/SRv6_Headend_behavioral_model.png "Figure 1: Headend Behavior ")
+__Figure 1: Headend Behavior.__
+
+### Transit Behavior ###
```

Comment:
Why is this same as Headend? Which exact behavior are your refering here ?

Reply:
It was defined in previous SRv6 SAI spec and I meant to say that transit behavior will be same as headend but I agree that it's doesn't make much sense here. I've removed this section.